### PR TITLE
adds new vault phase

### PR DESCRIPTION
**Pull request by drbh**
closes https://github.com/parallel-finance/parallel/issues/1037

### CODE REVIEW

File: pallets/crowdloans/src/lib.rs [hunk 1]
```diff
@@ -579,6 +579,7 @@ pub mod pallet {
             T::SlotExpiredOrigin::ensure_origin(origin)?;
 
             Self::try_mutate_vault(crowdloan, VaultPhase::Closed, |vault| {
+                ensure!(vault.phase == VaultPhase::Expired, Error::<T>::IncorrectPhase);
```
Comment (drbh) [inline pallets/crowdloans/src/lib.rs:582]: @GopherJ  should this be moved before of the `try_mutate_vault` callback?

**Comment by drbh** [comment]
@GopherJ when do we set the vault phase to `Succeed`?

File: pallets/crowdloans/src/lib.rs [hunk 2]
```diff
@@ -579,6 +579,13 @@ pub mod pallet {
         pub fn slot_expired(origin: OriginFor<T>, crowdloan: ParaId) -> DispatchResult {
             T::SlotExpiredOrigin::ensure_origin(origin)?;
 
+            // inital read to check if vault exists and is in the correct phase
+            let vault = Self::current_vault(crowdloan).ok_or(Error::<T>::VaultDoesNotExist)?;
+            ensure!(
```
Comment (GopherJ) [inline pallets/crowdloans/src/lib.rs:584]: this doesn't make sense because `expired` will need to come from `succeeded` now

Comment (GopherJ) [reply pallets/crowdloans/src/lib.rs:584]: and we will need to have a call to migrate from `Closed` to `Succeeded`

**Comment by GopherJ** [comment]
> 

in `auction_suceeded` call

**Comment by GopherJ** [comment]
@drbh let's add unit tests and bench item.

**Pull request merged**